app/components/errors/utils.ts

```diff
@@ -0,0 +1,39 @@
+import { z } from "zod";
+import { isRouteError } from "~/utils/http";
+
+export const specialErrorAdditionalData = z.discriminatedUnion("type", [
+  z.object({
+    type: z.literal("asset-from-other-org"),
+    assetOrganization: z.object({
+      organization: z.object({
+        id: z.string(),
+        name: z.string(),
+      }),
+    }),
+  }),
+]);
+
+export type SpecialErrorAdditionalData = z.infer<
+  typeof specialErrorAdditionalData
+>;
+
+export function parseSpecialErrorData(response: unknown):
+  | { success: false; additionalData: null }
```

Comment:
That is a very strange way to type it. Why not just `{success: boolean; additionalData: SpecialErrorAdditionalData | null}`